x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java

```diff
@@ -154,7 +206,7 @@ private static SpanProcessor createSpanProcessor(String endpoint, String token)
 
     public static class CapturingSpanExporter implements SpanExporter {
 
-        private List<SpanData> capturedSpans = new ArrayList<>();
+        private List<SpanData> capturedSpans = new CopyOnWriteArrayList<>();
```

Comment:
We're going to be writing to this quite a bit, I suggest `ConcurrentCollections.newQueue()` rather than `CopyOnWriteArrayList` here.

Reply:
Done